### PR TITLE
Freertos migration

### PR DESCRIPTION
**Pull request by mmhobi7**
Can someone take a look over this? I think it's straight forward. I assigned random ID's to the Messages. Have not tested on hardware

### CODE REVIEW

File: dev-board/Src/main.cpp [hunk 1]
```diff
@@ -312,27 +396,37 @@ int main(void)
...
+    /* USER CODE BEGIN Start_MC_Request */
+	RF_PACKET msg0{huart2.Instance};
+//	uint8_t rf_id=123;
+	bno055Init();
+    /* Infinite loop */
+    for(;;)
+    {
+  	  IMU_DATA_t imuData = bno055GetPacket();
+  	  IMU_TO_RF::AddMessage(&msg0, 123, &imuData); //TODO!! RF_ID!!
+  	  msg0.Send();
```
Comment (mmhobi7) [inline dev-board/Src/main.cpp]: TODO, move to rf transit packet thread

File: dev-board/Src/main.cpp [hunk 2]
```diff
@@ -423,7 +519,7 @@ static void MX_USART2_UART_Init(void)
 
   /* USER CODE END USART2_Init 1 */
   huart2.Instance = USART2;
-  huart2.Init.BaudRate = 57600;
+  huart2.Init.BaudRate = 38400;
```
Comment (mmhobi7) [inline dev-board/Src/main.cpp]: 57600

File: dev-board/Src/main.cpp [hunk 3]
```diff
@@ -108,159 +129,119 @@ int main(void)
...
+  		  msg0.Send();
+  	  }
+      osDelay(1);
+    }
+    //TODO! NEVER RUNS?!
+    ORION_MESSAGE_0_DATA_PACKET test{6.5343 , 6.5535, 1.6191, 43.29};
+    CAN_TO_RF::AddMessage(&msg0, RF_TYPES::ORION,334, &test); //TODO!! missing rf_id
+    test.packSummedVoltage = 52.32;
+    CAN_TO_RF::AddMessage(&msg0, RF_TYPES::ORION,335, &test); //TODO!! missing rf_id
+    msg0.Send();void StartDefaultTask(void const * argument);
```
Comment (mmhobi7) [inline dev-board/Src/main.cpp]: uh whoops

File: dev-board/Src/main.cpp [hunk 4]
```diff
@@ -312,27 +293,27 @@ int main(void)
 	if(!msg0.isPacketEmpty())
 		msg0.Send();
```
Comment (mmhobi7) [inline dev-board/Src/main.cpp]: Move into own thread ?

**Comment by Seth10001** [comment]
Have we come up with a flow chart of some kind for all the threads @John-Carr? That would likely be useful for setting this up. Looks like he's got the RTOS stuff up and running.

**Comment by mmhobi7** [comment]
devboard
I made a flowchart of the current code

File: .mxproject [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
-HeaderPath=Drivers\STM32F0xx_HAL_Driver\Inc;Drivers\STM32F0xx_HAL_Driver\Inc\Legacy;Drivers\CMSIS\Device\ST\STM32F0xx\Include;Drivers\CMSIS\Include;Inc;
-CDefines=USE_HAL_DRIVER;STM32F072xB;USE_HAL_DRIVER;STM32F072xB;
-
+[PreviousGenFiles]
```
Comment (mmhobi7) [inline .mxproject]: Line ending conversion from CR to CRLF (automatic by my git)

File: Src/main.cpp [hunk 6]
```diff
@@ -173,165 +161,160 @@ int main(void)
...
-  // request data from the motor controller
-  mcRequest.txData = { 1, 0, 1 };
-  mcRequest.SendData();
-  mppt0.SendData();
-  /* USER CODE END 2 */
-
-  /* Infinite loop */
-  /* USER CODE BEGIN WHILE */
-  while (1)
+  Start_CAN_Request(); // Send the first request packets for data from CAN
```
Comment (mmhobi7) [inline Src/main.cpp]: How will we handle timing of can request and can send?

**Comment by mmhobi7** [comment]
Ready for testing on hardware

**Pull request closed without merging**